### PR TITLE
7alip | Week3

### PR DESCRIPTION
**Pull request by 7alip**

### CODE REVIEW

File: Week3/homework/src/controllers/UsersController.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  };
+
+  getDetailedUsers = async (req, res, next) => {
+    try {
+      const { id } = req.params;
+      const details = await execQuery(userQueries.selectDetailedUsers());
+
+      let userId = [];
+
+      const result = details.reduce((arr, detail) => {
```
Comment (7alip) [inline Week3/homework/src/controllers/UsersController.js]: Is it possible to have the same result by only using SQL queries?
> **GET** /api/users/2/details
```json
[
    {
        "id": 2,
        "name": "patrick",
        "surname": "anthropy",
        "email": "panthropy@mail.com",
        "todos": [
            {
                "title": "Eating",
                "description": "You neede to drink 2lt water everyday",
                "tags": "important, immediately",
                "is_done": 0,
                "deadline": "2019-08-11T20:13:10.000Z",
                "remind_date": "2019-08-12T20:10:10.000Z"
            },
            {
                "title": "Sport",
                "description": "Dont miss that match!",
                "tags": "not required",
                "is_done": 0,
                "deadline": "2019-08-11T20:13:10.000Z",
                "remind_date": "2019-08-12T20:10:10.000Z"
            },
            {
                "title": "Eating",
                "description": "You neede to drink 2lt water everyday",
                "tags": "important, immediately",
                "is_done": 0,
                "deadline": "2019-08-10T22:00:00.000Z",
                "remind_date": "2019-08-11T22:00:00.000Z"
            },
            {
                "title": "Sport",
                "description": "Dont miss that match!",
                "tags": "not required",
                "is_done": 0,
                "deadline": "2019-08-10T22:00:00.000Z",
                "remind_date": "2019-08-11T22:00:00.000Z"
            }
        ]
    }
]
```

File: Week3/homework/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+## **TODO API**
+
+> < table > => users | todos | todo_lists | categories
+
+#### **Basic Crud Operations**
```
Comment (gajduk) [inline Week3/homework/README.md]: I like this 👍

File: Week3/homework/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+| ------------ | --------------------------- | :------------------------------------------------------------------------------------------------------------------------------ |
+| **`POST`**   | `/api/<table>/multiple`     | Create multiple users array<br>You must provide array with objects which contain custom properties                              |
+| **`DELETE`** | `/api/<table>/multiple/:id` | Delete users by multiple ids<br>Id numbers must be seperated by `|`<br>For example: `DELETE` => `/api/todos/multiple/1 | 2 | 4` |
+
+
+#### **Custom Operations**
+
+| Request   | Endpoints                  | Description                       |
+| --------- | -------------------------- | :-------------------------------- |
+| **`GET`** | `/api/user/:id/todos`      | Get all todos of a user           |
```
Comment (gajduk) [inline Week3/homework/README.md]: this is great. It is very common to see such patterns in rest where you can get a list of things for another object by following a route. It feels very natural 👍

File: Week3/homework/src/controllers/UsersController.js (hunk 1 above)
Comment (gajduk) [reply Week3/homework/src/controllers/UsersController.js]: I am afraid that is not possible with sql queries. You would need to use at least two queries one to get the user and another to get the todos and then combine the data.

**Comment by 7alip** [comment]
Hey @gajduk,
Thank you so much for your feedback and medal 😀
I knew something about ORMs. I had tried Entity Framework ORM while I was trying ASP.NET Core.
For NodeJS I read documentation about Sequelize. It is new for me to implement into homework. I may try ORM for next projects. Thanks for advice.

**Pull request closed without merging**